runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -7006,13 +7047,24 @@ J9::Z::TreeEvaluator::VMmonentEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    if (disableOOL)
       inlineRecursive = false;
 
+   if (TR::Compiler->om.areValueTypesEnabled() && isMonitorValueType == TR_maybe)
```

Comment:
Let's try to stick with similar code as on x86 and move this `TR_maybe` check into the API. Alternatively modify the x86 implementation to look like yours. Whichever you think is best as long as we are consistent.